sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ResolvedDataSource.scala

```diff
@@ -235,11 +236,18 @@ object ResolvedDataSource extends Logging {
         // For partitioned relation r, r.schema's column ordering can be different from the column
         // ordering of data.logicalPlan (partition columns are all moved after data column).  This
         // will be adjusted within InsertIntoHadoopFsRelation.
+
+        val codec = options.get("compression.codec").flatMap(e =>
+          Some(new CompressionCodecFactory(sqlContext.sparkContext.hadoopConfiguration)
+            .getCodecClassByName(e).asInstanceOf[Class[CompressionCodec]])
```

Comment:
Since user can only set compression through DataFrameWriter.compress(codec), it is very unlikely to set a unknown codec name. Even user set a unknown codec, I think it is fine to just throw exception on driver side.